### PR TITLE
net/bluetooth: Implement the start stop advertising

### PR DESCRIPTION
**Pull request by amitpurwar**
Signed-off-by: amit purwar <amit.purwar@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [6840832d215b05f8b534df9a34fc0263ef3eeb44] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6840832d215b05f8b534df9a34fc0263ef3eeb44] - Code Rule Check Result:
os/include/tinyara/bluetooth/bt_hci.h:63: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/bluetooth/bt_hci.h:63: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [ae4f3318c41944a46e1c6150cab2a4838265a598] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ae4f3318c41944a46e1c6150cab2a4838265a598] - Code Rule Check OK.

File: os/net/bluetooth/bluetooth.c [hunk 1]
```diff
@@ -311,6 +312,14 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_rand(void *buf, size_t len)
```
Comment (Wootak) [inline os/net/bluetooth/bluetooth.c]: Please add 'static' keyword.

File: os/net/bluetooth/bluetooth.c [hunk 2]
```diff
@@ -311,6 +312,14 @@ int bt_enable(bt_ready_cb_t cb)
...
+{
+	/*To check random.h: generate the random number */
+	/*getrandom(buf, len);*/
```
Comment (Wootak) [inline os/net/bluetooth/bluetooth.c]: To use getrandom() function, we need to enable below 2 configs.
CONFIG_CRYPTO, CONFIG_CRYPTO_RANDOM_POOL

File: os/net/bluetooth/bluetooth.c [hunk 3]
```diff
@@ -523,21 +532,358 @@ int bt_id_delete(uint8_t id)
...
+	err = bt_rand(nrpa.val, sizeof(nrpa.val));
+	if (err) {
+		return err;
+	}
+
+	nrpa.val[5] &= 0x3f;
+	return set_random_address(&nrpa);
+}
+
+int bt_le_adv_start_internal(const struct bt_le_adv_param *param, const struct bt_data *ad, size_t ad_len, const struct bt_data *sd, size_t sd_len, const bt_addr_le_t *peer)
```
Comment (Wootak) [inline os/net/bluetooth/bluetooth.c]: Please add 'static' keyword.

**Comment by seinfra** [comment]
Target : [6fd90fb24914e8fe9abb5db33dca42ed2cb53a44] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6fd90fb24914e8fe9abb5db33dca42ed2cb53a44] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e552ab3efc849069f6a7ba140eca54c3409d8978] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e552ab3efc849069f6a7ba140eca54c3409d8978] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0a9bcf4b7845e390e1530dd6cdc338d79314727e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0a9bcf4b7845e390e1530dd6cdc338d79314727e] - Code Rule Check (C++) OK.

**Pull request merged**